include/oneapi/dpl/internal/dynamic_selection_impl/fixed_resource_policy.h

```diff
@@ -0,0 +1,127 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_STATIC_POLICY_IMPL_H
+#define _ONEDPL_STATIC_POLICY_IMPL_H
+#include <vector>
+#include <type_traits>
+#include <memory>
+#include <exception>
+#include "oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h"
+#if _DS_BACKEND_SYCL != 0
+    #include "oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h"
+#endif
+namespace oneapi {
+namespace dpl {
+namespace experimental {
+
+#if _DS_BACKEND_SYCL != 0
+  template <typename Backend=sycl_backend>
+#else
+  template <typename Backend>
+#endif
+  struct fixed_resource_policy {
+  private:
+    using backend_t = Backend;
+    using resource_container_t = typename backend_t::resource_container_t;
+    using execution_resource_t = typename backend_t::execution_resource_t;
+    using wrapped_resource_t = typename std::decay_t<Backend>::execution_resource_t;
+
+  public:
+    //policy traits
+    using resource_type = decltype(unwrap(std::declval<wrapped_resource_t>()));
+    using selection_type = oneapi::dpl::experimental::basic_selection_handle_t<fixed_resource_policy<Backend>, execution_resource_t>;
+    using wait_type = typename backend_t::wait_type;
+
+  private:
+    std::shared_ptr<backend_t> backend_;
+
+    struct state_t {
+        resource_container_t resources_;
+        int offset_;
+    };
+
+    std::shared_ptr<state_t> state_;
+
+  public:
+    fixed_resource_policy(int offset=0) {
+        if(offset!=deferred_initialization){
+            initialize(offset);
+        }
+    }
+
+    fixed_resource_policy(const std::vector<resource_type>& u, int offset=0) {
+        if(offset!=deferred_initialization){
+            initialize(u, offset);
+        }
+    }
+
+    auto get_resources()  const {
+      if(backend_){
+          return backend_->get_resources();
+      }else{
+          throw std::runtime_error("Called select before initialization\n");
```

Comment:
The message is wrong, as the called method is `get_resources`, not `select`

Reply:
Will be fixed.

Reply:
Fixed